### PR TITLE
fix: istio 1.14.3 only delete gateway resources associated with specific istiooperator CR

### PR DESCRIPTION
**Pull request by chinaran**
**Please provide a description of this PR:**

Please refer to: https://github.com/istio/istio/issues/40577

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [x] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by holla-yangchunping — APPROVED** [review]

**Review by Chen-Xintong — APPROVED** [review]
lgtm!

**Review by howardjohn — COMMENTED** [review]
Do we have any tests for this? Its not clear why the logic is gateway specific to me

**Comment by chinaran** [comment]
@howardjohn I added unit test, please check out.

**Comment by zirain** [comment]
> Adding the label to cherry-pick to 1.14.

also need cherrypick to 1.15? maybe 1.13?

**Comment by chinaran** [comment]
@howardjohn Is there anything unclear?

**Comment by chinaran** [comment]
@Chen-Xintong @ericvn @zirain @howardjohn What else needs to be done to merge this PR into master?

**Comment by ericvn** [comment]
Since @howardjohn had a question, I'll leave this for his approval. The stale label should bereaved with this comment. I'll also kick the one test that hasn't completed.

**Comment by chinaran** [comment]
@ericvn Thank you for your efforts.

**Comment by ericvn** [comment]
@istio/wg-environments-maintainers (@howardjohn is a member of that group) still need to approve this.

**Comment by auswells** [comment]
We are running Istio-Operator 1.14.4 and have come across the issue this PR is addressing. I've tested the changes from this PR in one of our clusters and can confirm that IOP deletions no longer delete unrelated IOP gateway resources. If there's anything I can do to assist in getting this released let me know.

**Comment by ericvn** [comment]
@howardjohn Thoughts on this?

**Comment by ericvn** [comment]
Also going to remove the releasenotes-none label as we should have a release notes propogated through the releases if this is needed.

**Comment by zirain** [comment]
@chinaran can you update with a release-notes?

**Review by zirain — APPROVED** [review]
lgtm

File: releasenotes/notes/40578.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: installation
+
+issue:
+  - 40577
+
+releaseNotes:
+  - |
+    **Fixed** an issue when user deletes a custom gateway by iop CR, other gateways are restarted.
```
Comment (ericvn) [inline releasenotes/notes/40578.yaml:10]:
```suggestion
    **Fixed** an issue where deleting a custom gateway using a Istio Operator custom resource, other gateways are restarted.
```

**Review by ericvn — APPROVED** [review]
Test and release notes look good. Thanks.

**Pull request merged**